src/plasma-cash/entity.ts

```diff
@@ -562,4 +596,20 @@ export class Entity {
       gas: this._defaultGas
     })
   }
+
+  /**
+   * 
+   * @param tx The transaction's receipt that wea want to decode
+   * @param i The Deposit event is the i'th emitted event. Set to 0 for depositing ETH, Set to 1 for ERC20/ERC721 since the first event is a `Transfer` event
+   */
+  async logParser(tx: any, i: number): Promise<IPlasmaCoin> {
```

Comment:
Maybe make this `getCoinFromTxAsync(txHash: string, i: number): Promise<IPlasmaCoin>`?